sdk/typescript/src/signers/txn-data-serializers/call-arg-serializer.ts

```diff
@@ -154,12 +154,19 @@ export class CallArgSerializer {
     return hasTxContext ? params.slice(0, params.length - 1) : params;
   }
 
-  async newObjectArg(objectId: string): Promise<ObjectArg> {
+  async newObjectArg(
+    objectId: string,
+    expectedType: SuiMoveNormalizedType,
+  ): Promise<ObjectArg> {
     const object = await this.provider.getObject(objectId);
     const initialSharedVersion = getSharedObjectInitialVersion(object);
 
-    const mutable = true; // Defaulted to True to match current behavior.
     if (initialSharedVersion) {
+      const mutable =
+        typeof expectedType === 'object' && 'MutableReference' in expectedType
+          ? true
+          : false;
+
```

Comment:
```suggestion
      const mutable =
        typeof expectedType === 'object' && 'MutableReference' in expectedType;
```

Reply:
@andll / @amnn is the backend ready for testing this change? e.g., having a test case where if the `mutable` is set to a wrong value, the move call will fail

Reply:
Currently, the only case it will fail for is `Clock` (found at address `0x6`), but yes, that is all set-up on `main`, and should be going out to devnet with v0.27.0 (today).

Reply:
In addition to Clock check, we also do have a check that makes sure that you can't pass argument that has `mutable: false` to a function that takes `&mut` argument